test/unit/util/test_cuda_sparse_matrix_vector_ordering.cpp

```diff
@@ -0,0 +1,230 @@
+#include <gtest/gtest.h>
+
+#include <micm/util/cuda_sparse_matrix.hpp>
+#include <micm/util/sparse_matrix_vector_ordering.hpp>
+
+#include "test_sparse_matrix_policy.hpp"
+#include "cuda_matrix_utils.cuh"
+
+TEST(CudaSparseVectorMatrix, ZeroMatrix)
+{
+  auto matrix = testZeroMatrix<micm::CudaSparseMatrix, micm::SparseMatrixVectorOrdering<2>>();
+
+  matrix.CopyToDevice();
+  auto param = matrix.AsDeviceParam();
+  micm::cuda::AddOneDriver(param);
+
+  for(auto& elem : matrix.AsVector())
+  {
+    EXPECT_EQ(elem, 1.0);
+  }
+
+  matrix.CopyToHost();
+
+  for(auto& elem : matrix.AsVector())
+  {
+    EXPECT_EQ(elem, 1.0);
+  }
+}
+
+TEST(CudaSparseVectorMatrix, CopyAssignmentConstZeroMatrix)
+{
+  const auto matrix = testConstZeroMatrix<micm::CudaSparseMatrix, micm::SparseMatrixVectorOrdering<3>>();
+  auto oneMatrix = matrix;
+  oneMatrix.CopyToDevice();
+  auto param = oneMatrix.AsDeviceParam();
+  micm::cuda::AddOneDriver(param);
+  
+  for(auto& elem : matrix.AsVector())
+  {
+    EXPECT_EQ(elem, 0.0);
+  }
+  for(auto& elem : oneMatrix.AsVector())
+  {
+    EXPECT_EQ(elem, 0.0);
+  }
+
+  oneMatrix.CopyToHost();
+
+  for(auto& elem : matrix.AsVector())
+  {
+    EXPECT_EQ(elem, 0.0);
+  }
+  for(auto& elem : oneMatrix.AsVector())
+  {
+    EXPECT_EQ(elem, 1.0);
+  }
+}
+
+TEST(CudaSparseVectorMatrix, CopyAssignmentDeSynchedHostZeroMatrix)
+{
+  auto matrix = testZeroMatrix<micm::CudaSparseMatrix, micm::SparseMatrixVectorOrdering<3>>();
+  matrix.CopyToDevice();
+  auto param = matrix.AsDeviceParam();
+  micm::cuda::AddOneDriver(param);
+
+  for(auto& elem : matrix.AsVector())
+  {
+    EXPECT_EQ(elem, 0.0);
+  }
+
+  auto oneMatrix = matrix;
+
+  for(auto& elem : matrix.AsVector())
+  {
+    EXPECT_EQ(elem, 0.0);
+  }
+  for(auto& elem : oneMatrix.AsVector())
+  {
+    EXPECT_EQ(elem, 0.0);
+  }
+
+  matrix.CopyToHost();
+
+  for(auto& elem : matrix.AsVector())
+  {
+    EXPECT_EQ(elem, 1.0);
+  }
+  for(auto& elem : oneMatrix.AsVector())
+  {
+    EXPECT_EQ(elem, 0.0);
+  }
+
+  oneMatrix.CopyToHost();
+
+  for(auto& elem : matrix.AsVector())
+  {
+    EXPECT_EQ(elem, 1.0);
+  }
+  for(auto& elem : oneMatrix.AsVector())
+  {
+    EXPECT_EQ(elem, 1.0);
+  }
+}
+
+TEST(CudaSparseVectorMatrix, MoveAssignmentConstZeroMatrix)
+{
+  const auto matrix = testConstZeroMatrix<micm::CudaSparseMatrix, micm::SparseMatrixVectorOrdering<3>>();
+  auto oneMatrix = std::move(matrix);
```

Comment:
it might be good to have some device-side data on `matrix` prior to the move, to make sure it gets moved correctly

Reply:
I believe we are doing that in `MoveAssignmentDeSyncedHostZeroMatrix`?